lib/lm_docstache/hide_custom_tags.rb

```diff
@@ -0,0 +1,65 @@
+module LMDocstache
+  class HideCustomTags
+    HIDE_BACKGROUND_COLOR = 'FFFFFF'
+
+    attr_reader :document, :hide_custom_tags
+    def initialize(document:, hide_custom_tags: [])
+      @document = document
+      @hide_custom_tags = hide_custom_tags
+    end
+
+    def hide_custom_tags!
+      hide_custom_tags.each do |full_pattern|
+        paragraphs = document.css('w|p')
+        while paragraph = paragraphs.shift do
+          next unless paragraph.text =~ full_pattern
+          run_nodes = paragraph.css('w|r')
+          while run_node = run_nodes.shift
+            next if run_node.text.to_s.strip.size == 0
+            remainder_run_node = run_node.clone
+            run_node.unlink
+            tag_contents = split_tag_content(remainder_run_node.text, full_pattern)
+            tag_contents[:content_list].each_with_index do |content, idx|
+              replace_content(remainder_run_node, content)
+              run_node_with_match = remainder_run_node.dup
+              matched_tags = tag_contents[:matched_tags][idx]
+              nodes_list = [remainder_run_node]
+              if matched_tags
+                replace_style(run_node_with_match)
+                replace_content(run_node_with_match, matched_tags)
+                nodes_list << run_node_with_match
+              end
+              paragraph << Nokogiri::XML::NodeSet.new(document, nodes_list)
+              remainder_run_node = remainder_run_node.clone
+            end
+          end
+        end
+      end
+    end
+
+    private
+    def split_tag_content(text, full_pattern)
+      content_list = text.split(full_pattern)
```

Comment:
It's important to remember that for your logic to work with `String#split`, the full_pattern must be a `Regexp` without capture groups.

Reply:
I'm not sure what you mean. Do you want me to describe this as a documentation?

https://github.com/boost-legal/lm-docstache/pull/8/files#diff-473de67f52e8aca33647ef59951c5e059ee4ed9ab5fdf9287a121332db52fbbbR46

Reply:
I think we should at least leave this as comments somewhere in the code.